polynote-spark/src/main/scala/polynote/kernel/LocalSparkKernel.scala

```diff
@@ -153,11 +153,14 @@ class LocalSparkKernelFactory extends Kernel.Factory.LocalService {
       if (conf.get("spark.master") == "local[*]") {
         conf.set("spark.driver.host", "127.0.0.1")
       }
+      
+      if (conf.get("spark.app.name", null) == null) {
+        conf.setAppName(s"Polynote ${BuildInfo.version} session")
+      }
```

Comment:
@MehdiFal Thanks so much for the PR! 

I wonder whether it'd be a bit cleaner to use `conf.setIfMissing`? That way we don't need this if statement at all, or any ugly `null`s :)

Reply:
Was about to put up a PR for this and then realised that @MehdiFal beat me to it :)
You can also do something like this to avoid nulls
```
      conf
        .setAppName(
          conf.getOption("spark.app.name") match {
            case Some(appName) if appName.length > 0 => appName
            case _ => s"Polynote ${BuildInfo.version} session"
          }
        )
```
But what @jonathanindig suggesting is more consistent with the rest of the file.